From b2d272f3df64c2372e40302fb0c6d8e066862c40 Mon Sep 17 00:00:00 2001
From: Elizabeth Figura <zfigura@codeweavers.com>
Date: Mon, 29 Dec 2025 17:36:30 -0600
Subject: [PATCH] ddraw: Fix caps tracing in IDirectDraw7::GetCaps().

The output pointers don't necessarily point to a v7 caps structure,
and we currently may trace invalid members, e.g. ddsCaps.

(cherry picked from commit c54132893fc7e0ce3544ff739b727059f79214fe)
---
 dlls/ddraw/ddraw.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/dlls/ddraw/ddraw.c b/dlls/ddraw/ddraw.c
index 6028a8463ef..1fe6a101541 100644
--- a/dlls/ddraw/ddraw.c
+++ b/dlls/ddraw/ddraw.c
@@ -1594,9 +1594,8 @@ static HRESULT WINAPI ddraw7_GetCaps(IDirectDraw7 *iface, DDCAPS *DriverCaps, DD
         if (TRACE_ON(ddraw))
         {
             TRACE("Driver Caps :\n");
-            DDRAW_dump_DDCAPS(DriverCaps);
+            DDRAW_dump_DDCAPS(&caps);
         }
-
     }
     if(HELCaps)
     {
@@ -1604,7 +1603,7 @@ static HRESULT WINAPI ddraw7_GetCaps(IDirectDraw7 *iface, DDCAPS *DriverCaps, DD
         if (TRACE_ON(ddraw))
         {
             TRACE("HEL Caps :\n");
-            DDRAW_dump_DDCAPS(HELCaps);
+            DDRAW_dump_DDCAPS(&caps);
         }
     }
 
